src/adapter/src/metrics.rs

```diff
@@ -27,6 +27,7 @@ pub struct Metrics {
     pub storage_usage_collection_time_seconds: HistogramVec,
     pub subscribe_outputs: IntCounterVec,
     pub canceled_peeks: IntCounterVec,
+    pub persist_peek_seconds: Histogram,
```

Comment:
I'm not sure how adapter likes to structure metrics... happy to take advice here!

Reply:
https://materializeinc.slack.com/archives/C02FWJ94HME/p1694805197123109 I asked in the surfaces channel, we've just recently been thinking about questions like this, but I'm not sure if we have a concrete answer yet.

Reply:
We don't have many opinions yet. I think it's fine as written here and we can refactor if desired?